src/compile/scale.ts

```diff
@@ -504,3 +506,78 @@ export function zero(scale: Scale, channel: Channel, fieldDef: FieldDef) {
   }
   return undefined;
 }
+
+/**
+ * Merge scales from children by unioning their domains
+ */
+export function mergeScales(model: LayerModel, channel: string) {
+  let scaleComp: ScaleComponents = null;
+
+  let domains = {};
+
+  model.children().forEach((child) => {
+    const childScales = child.component.scale[channel];
+    if (!childScales) {
+      return;
+    }
+
+    // range and other things will just be defined by the first scale
```

Comment:
This is not always true yet, given than you only determine whether to merge by `type`.   

We need to settle that first, and that will determine the correct rule here.